### PR TITLE
Disallow netmasks in devices' ip to prevent overlaps in firewall

### LINKED ISSUE #1139: Prevent subnet in device IPs
Prevent devices from being created with CIDRs for IPs (`/24` etc).

Refs #1135

### PR DESCRIPTION
**Pull request by conectado**
Fixes #1139 

/32 is still allowed for ipv4 and /128 is still allowed for ipv6

I think no migration is needed since netmasks in device ips were broken before(Look at the address field):

image

The error looks like:

image

### CODE REVIEW

File: apps/fz_http/lib/fz_http/devices/device.ex [hunk 1]
```diff
@@ -225,6 +247,14 @@ defmodule FzHttp.Devices.Device do
 
   defp validate_in_network(changeset, :ipv6), do: changeset
 
+  defp netmask_error(changeset, ip_type) do
+    add_error(
+      changeset,
+      ip_type,
+      "Only IPs without netmask are supported, try using the Allowed IPs field for routing within the network"
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http/devices/device.ex]: I think we want to leave out the AllowedIPs note since we don’t support that just yet.

File: apps/fz_http/lib/fz_http/devices/device.ex [hunk 2]
```diff
@@ -211,6 +213,26 @@ defmodule FzHttp.Devices.Device do
     end
   end
 
+  defp validate_single_ip(%Ecto.Changeset{changes: %{ipv4: %{netmask: 32}}} = changeset, :ipv4),
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http/devices/device.ex]: Prob want to disallow specifying the mask because I think we append /32 and /128 currently

Comment (jamilbk) [reply apps/fz_http/lib/fz_http/devices/device.ex]: Then you can just check for nil I think to pass validation, or error if the netmask exists

File: apps/fz_http/lib/fz_http/devices/device.ex [hunk 3]
```diff
@@ -211,6 +213,26 @@ defmodule FzHttp.Devices.Device do
...
+    do: changeset
+
+  defp validate_single_ip(%Ecto.Changeset{changes: %{ipv4: %{netmask: nil}}} = changeset, :ipv4),
```
Comment (jamilbk) [inline apps/fz_http/lib/fz_http/devices/device.ex]: Nit: naming: `validate_no_mask`

**Review by jamilbk — CHANGES_REQUESTED** [review]
Nice approach, just a few minor things

**Comment by jamilbk** [comment]
The build error should be fixed by a rebase

File: apps/fz_http/lib/fz_http/devices/device.ex (hunk 2 above)
Comment (conectado) [reply apps/fz_http/lib/fz_http/devices/device.ex]: I tested it and it works correctly with /32 or /128 no double slash. Maybe postgres inet deletes the slash? Will erase it so we don't depend on an undocumented feature.

Comment (conectado) [reply apps/fz_http/lib/fz_http/devices/device.ex]: It seems that the /32 or /128 are automatically added when the changeset is created(but not when migrations run) so I need to keep both the test for /32 and `nil`

**Comment by conectado** [comment]
@jamilbk weird mix format is failing but it's passing locally

**Comment by jamilbk** [comment]
@conectado Hmm let me add a test real quick and do the mix format on my machine. Mind if I push up a commit?

**Comment by conectado** [comment]
> @conectado Hmm let me add a test real quick and do the mix format on my machine. Mind if I push up a commit?

Go ahead!

**Comment by jamilbk** [comment]
But yeah that's weird it's failing in the pipeline but not locally. `pre-commit run --all-files` is run in CI so it may be catching something else.

**Review by jamilbk — DISMISSED** [review]

**Review by jamilbk — DISMISSED** [review]
